api/cloud/oc_cloud_manager.c

```diff
@@ -393,18 +415,28 @@ cloud_login(void *data)
   return OC_EVENT_DONE;
 }
 
+static oc_cloud_error_t
+_refresh_token_handler_check_data_error(oc_client_response_t *data)
+{
+  if (data->code >= OC_STATUS_SERVICE_UNAVAILABLE) {
+    return CLOUD_ERROR_CONNECT;
+  }
+  if (data->code >= OC_STATUS_BAD_REQUEST) {
+    return CLOUD_ERROR_REFRESH_ACCESS_TOKEN;
+  }
+  return CLOUD_OK;
+}
+
 static int
 _refresh_token_handler(oc_cloud_context_t *ctx, oc_client_response_t *data)
 {
-  if (data->code >= OC_STATUS_SERVICE_UNAVAILABLE) {
-    cloud_set_last_error(ctx, CLOUD_ERROR_CONNECT);
-    goto error;
-  } else if (data->code >= OC_STATUS_BAD_REQUEST) {
-    cloud_set_last_error(ctx, CLOUD_ERROR_REFRESH_ACCESS_TOKEN);
+  oc_cloud_error_t err = _refresh_token_handler_check_data_error(data);
+  if (err != CLOUD_OK) {
+    cloud_set_last_error(ctx, err);
```

Comment:
pls remove the line - in `error` space set it via `cloud_set_cps_and_last_error`